### ISSUE TITLE
pre-deployment-checklist: a sub-check defers to a waiver but not to the owning law passing

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#107]
**Version:** 7.21.0
**Law:** `pre-deployment-checklist-mandatory`

Follow-on to #96, which you fixed: sub-checks now defer to `notApplicable`. This
is the other half of the same idea — a sub-check that still fails while the law
owning the concern **passes**.

## What happens

```
pre-deployment-checklist-mandatory → "Logging not properly configured"
centralized-logging                → passed: true, score 100
```

Same audit, same run. The checklist's own analyzer decides:

```js
// laws/deployment/pre-deployment-checklist.js:81
const loggingConfigured = monitoringSetup.loggingConfigured ||
  PythonSatisfaction.hasStructuredLogging(projectRoot);
if (!loggingConfigured && !concernIsWaived(context.config, 'Centralized Logging')) {
  violations.push('Logging not properly configured');
}
```

The escape hatch is a **waiver**. There is no branch for "the law that owns this
concern ran and passed", so a project that genuinely satisfies `Centralized
Logging` is told by the checklist that it has not.

The health-check sub-check immediately above behaves correctly here only because
we happen to have waived `Health Check Monitoring` — had we satisfied it instead,
it would report the same way.

## Why it matters more after #96

#96 made these sub-checks respect an adopter's written `notApplicable`. That
closed the case where the concern does not exist. The remaining case is the
better one: the concern exists, the project meets it, the owning law says so, and
the checklist disagrees.

It also produces the perverse incentive of waiving a law you pass, purely so a
different law stops reporting it.

## Suggested direction

Where a sub-check names a concern that is also a law, resolve it in this order:

1. the concern is waived → satisfied (today's behaviour, from #96)
2. **the owning law ran and passed → satisfied** (missing)
3. otherwise → run the local analyzer

The report already carries what is needed; `centralized-logging` is in the
results with `passed: true`. The same pattern applies to any other sub-check
paired with a law — health checks, authentication, performance benchmarking.

Worth a sweep: #96's fix and this one are the same question asked twice, and the
ratio in your own release notes suggests there will be more than these two.
